examples/EL-attention/summarize.py

```diff
@@ -0,0 +1,120 @@
+                hypotheses_batch = bart.sample(slines, **eval_kwargs)
+                for hypothesis in hypotheses_batch:
+                    fout.write(hypothesis + "\n")
+                    fout.flush()
+                slines = []
+
+            slines.append(sline.strip())
+            count += 1
+
+        if slines != []:
+            hypotheses_batch = bart.sample(slines, **eval_kwargs)
+            for hypothesis in hypotheses_batch:
+                fout.write(hypothesis + "\n")
+                fout.flush()
+
+
+def main():
+    """
+    Usage::
+         python examples/bart/summarize.py \
+            --model-dir $HOME/bart.large.cnn \
+            --model-file model.pt \
+            --src $HOME/data-bin/cnn_dm/test.source
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--model-dir",
+        required=True,
+        type=str,
+        default="bart.large.cnn/",
+        help="path containing model file and src_dict.txt",
+    )
+    parser.add_argument(
+        "--model-file",
+        default="checkpoint_best.pt",
+        help="where in model_dir are weights saved",
+    )
+    parser.add_argument(
+        "--src", default="test.source", help="text to summarize", type=str
+    )
+    parser.add_argument(
+        "--out", default="test.hypo", help="where to save summaries", type=str
+    )
+    parser.add_argument("--bsz", default=32, help="where to save summaries", type=int)
+    parser.add_argument(
+        "--n", default=None, help="how many examples to summarize", type=int
+    )
+    parser.add_argument(
+        "--xsum-kwargs",
+        action="store_true",
+        default=False,
+        help="if true use XSUM_KWARGS else CNN_KWARGS",
+    )
+    # Fastseq related setting
+    parser.add_argument('--use-fastseq', action='store_true',
+            help='Use fastseq optimization ? ')
+    parser.add_argument('--use-el-attn', action='store_true',
+            help='Use Efficient Lossless Attention optimization ? ')
+    args = parser.parse_args()
+    
+    # Check Fastseq related setting
+    os.environ['USE_EL_ATTN'] = '1' if args.use_el_attn else '0'
+    if args.use_el_attn or args.use_fastseq:
+        import fastseq
+
+    from fairseq.models.bart import BARTModel
+
+    eval_kwargs = XSUM_KWARGS if args.xsum_kwargs else CNN_KWARGS
+    if args.model_dir == "pytorch/fairseq":
+        bart = torch.hub.load("pytorch/fairseq", args.model_file)
+    else:
+        bart = BARTModel.from_pretrained(
+            args.model_dir,
+            checkpoint_file=args.model_file,
+            data_name_or_path=args.model_dir,
+        )
+    bart = bart.eval()
+    if torch.cuda.is_available():
+        bart = bart.cuda().half()
+
+    # TODO make this step automatic
+    if args.use_el_attn:
+        #for model in bart.models:
+        bart.model.transpose_enc_dec_kv_proj()
+        #bart.model.set_beam_size(args.bsz)
+        #param = dict({'beamable_mm_beam_size': args.bsz})
```

Comment:
Remove comments and extra blank lines. Remove #TODO is already done.

Reply:
removed comments. TODO is not done, keep it here.